includes/Classifai/Features/TitleGeneration.php

```diff
@@ -95,6 +95,61 @@ public function feature_setup() {
 		add_action( 'edit_form_before_permalink', [ $this, 'register_generated_titles_template' ] );
 	}
 
+	/**
+	 * Register the ability for the Feature.
+	 */
+	public function register_ability() {
+		wp_register_ability(
+			'classifai/generate-title',
+			[
+				'label'               => esc_html__( 'Generate title suggestions', 'classifai' ),
+				'description'         => esc_html__( 'Use AI to generate title suggestions based on content. This can be content that is passed in or content that is pulled from a post via the post ID. Will return an array of title suggestions.', 'classifai' ),
+				'input_schema'        => [
+					'type'       => 'object',
+					'properties' => [
+						'content' => [
+							'type'              => 'string',
+							'sanitize_callback' => 'sanitize_text_field',
+							'description'       => esc_html__( 'Content to generate a title for.', 'classifai' ),
+						],
+						'id'      => [
+							'type'              => 'integer',
+							'sanitize_callback' => 'absint',
+							'description'       => esc_html__( 'Post ID to generate a title for.', 'classifai' ),
+						],
+						'n'       => [
+							'type'              => 'integer',
+							'minimum'           => 1,
+							'maximum'           => 10,
+							'sanitize_callback' => 'absint',
+							'description'       => esc_html__( 'Number of titles to generate', 'classifai' ),
+						],
+					],
+					'required'   => [
+						'id',
+					],
+				],
+				'output_schema'       => [
+					'type'       => 'object',
```

Comment:
In my testing, seems we need to return an object in order to support any `WP_Error`'s that may return. This does mean we have to change the structure of what we return to be more than just a simple array unfortunately